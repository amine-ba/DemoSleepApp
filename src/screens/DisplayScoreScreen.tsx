import React from "react";
import { Button, Heading, Progress, Text } from "native-base";

import theme from "@shared/theme";
import { FormContainer, SpacingContainer } from "@shared/styles";

import { TextContent } from "@consts";

export const DisplayScoreScreen = ({ score, reset }: DailyScoreScreenProps) => {
  return (
    <FormContainer safeAreaTop>
      <SpacingContainer mVertical={50}>
        <Heading>{TextContent.scoreScreen.title}</Heading>
      </SpacingContainer>

      <Text fontSize="5xl" alignSelf={"center"} testID="display.score">
        {score} %
      </Text>
      <Progress size="2xl" value={score} />

      <SpacingContainer mTop={50}>
        <Button
          testID="reset.score.botton"
          colorScheme={theme.colors.primary.red}
          color={theme.colors.primary.red300}
          variant="subtle"
          onPress={reset}
        >
          {TextContent.reset}
        </Button>
      </SpacingContainer>
    </FormContainer>
  );
};

export interface DailyScoreScreenProps {
  reset: () => void;
  score: number;
}

export default DisplayScoreScreen;
