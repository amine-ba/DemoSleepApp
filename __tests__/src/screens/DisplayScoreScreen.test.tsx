import React from "react";
import { render, fireEvent } from "@testing-library/react-native";

import DisplayScoreScreen, {
  DailyScoreScreenProps,
} from "@screens/DisplayScoreScreen";
import { NativeBaseProvider } from "native-base";

const inset = {
  frame: { x: 0, y: 0, width: 0, height: 0 },
  insets: { top: 0, left: 0, right: 0, bottom: 0 },
};

const Mockrest = jest.fn();
let testScore = 1;

const TestComponent = (props: DailyScoreScreenProps) => (
  <NativeBaseProvider initialWindowMetrics={inset}>
    <DisplayScoreScreen {...props} />
  </NativeBaseProvider>
);

describe(" <DisplayScoreScreen /> Component ", () => {
  it(" Should Successfully Rendred ", () => {
    const json = render(<TestComponent score={1} reset={Mockrest} />).toJSON();

    expect(json).toMatchSnapshot();
  });

  it(" Should Display the exact Score ", async () => {
    const { findByTestId } = render(
      <TestComponent score={1} reset={Mockrest} />
    );

    const displayChild = await findByTestId("display.score");

    expect(displayChild?.props?.children?.join("")).toEqual(`${testScore} %`);
  });

  it(" Should Call Reset ", async () => {
    const { findByTestId } = render(
      <TestComponent score={1} reset={Mockrest} />
    );

    const resetButtonNode = await findByTestId("reset.score.botton");

    fireEvent.press(resetButtonNode);

    // only once
    expect(Mockrest).toBeCalledTimes(1);
  });
});
