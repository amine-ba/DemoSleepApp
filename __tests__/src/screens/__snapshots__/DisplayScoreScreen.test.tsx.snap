// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` <App /> Component   Should Successfully Rendred  1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    className="sc-bdvvtL sc-gsDKAQ  giHPxp"
    dataSet={Object {}}
    style={Object {}}
  >
    <View
      className="sc-dkPtRN bZylFl"
      dataSet={Object {}}
      mVertical={50}
      style={Object {}}
    >
      <Text
        dataSet={Object {}}
        style={
          Object {
            "backgroundColor": undefined,
            "color": "#404040",
            "fontFamily": undefined,
            "fontSize": null,
            "fontStyle": "normal",
            "fontWeight": "700",
            "letterSpacing": 0,
            "lineHeight": 30,
            "textDecorationLine": undefined,
          }
        }
      >
        Your Score is
      </Text>
    </View>
    <Text
      dataSet={Object {}}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": undefined,
          "color": "#1f2937",
          "fontFamily": undefined,
          "fontSize": 48,
          "fontStyle": "normal",
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 72,
          "textDecorationLine": undefined,
        }
      }
      testID="display.score"
    >
      1
       %
    </Text>
    <View
      accessibilityRole="progressbar"
      accessibilityValue={
        Object {
          "max": 100,
          "min": 0,
          "now": 1,
        }
      }
      accessible={true}
      colorScheme="primary"
      dataSet={Object {}}
      isIndeterminate={false}
      style={
        Object {
          "backgroundColor": "#a5f3fc",
          "borderRadius": 9999,
          "height": 24,
          "overflow": "hidden",
        }
      }
    >
      <View
        dataSet={Object {}}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#0891b2",
            "display": "flex",
            "elevation": 1,
            "height": "100%",
            "justifyContent": "center",
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.18,
            "shadowRadius": 1,
            "width": "1%",
          }
        }
      />
    </View>
    <View
      className="sc-dkPtRN ioVHrU"
      dataSet={Object {}}
      mTop={50}
      style={Object {}}
    >
      <View
        _text={
          Object {
            "color": "darkText",
          }
        }
        accessibilityRole="button"
        accessible={true}
        colorScheme="red"
        dataSet={Object {}}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onHoverIn={[Function]}
        onHoverOut={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fee2e2",
            "borderRadius": 4,
            "color": "#fca5a5",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingBottom": 8,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 8,
          }
        }
        testID="reset.score.botton"
      >
        <View
          dataSet={Object {}}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            dataSet={Object {}}
            style={Object {}}
          >
            <Text
              dataSet={Object {}}
              style={
                Object {
                  "backgroundColor": undefined,
                  "color": "#ef4444",
                  "fontFamily": undefined,
                  "fontSize": 14,
                  "fontStyle": "normal",
                  "fontWeight": "500",
                  "letterSpacing": 0,
                  "lineHeight": 21,
                  "textDecorationLine": undefined,
                }
              }
            >
              Reset
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[` <App /> Component   should successfully rendred  1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    className="sc-bdvvtL sc-gsDKAQ  giHPxp"
    dataSet={Object {}}
    style={Object {}}
  >
    <View
      className="sc-dkPtRN bZylFl"
      dataSet={Object {}}
      mVertical={50}
      style={Object {}}
    >
      <Text
        dataSet={Object {}}
        style={
          Object {
            "backgroundColor": undefined,
            "color": "#404040",
            "fontFamily": undefined,
            "fontSize": null,
            "fontStyle": "normal",
            "fontWeight": "700",
            "letterSpacing": 0,
            "lineHeight": 30,
            "textDecorationLine": undefined,
          }
        }
      >
        Your Score is
      </Text>
    </View>
    <Text
      dataSet={Object {}}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": undefined,
          "color": "#1f2937",
          "fontFamily": undefined,
          "fontSize": 48,
          "fontStyle": "normal",
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 72,
          "textDecorationLine": undefined,
        }
      }
      testID="display.score"
    >
      1
       %
    </Text>
    <View
      accessibilityRole="progressbar"
      accessibilityValue={
        Object {
          "max": 100,
          "min": 0,
          "now": 1,
        }
      }
      accessible={true}
      colorScheme="primary"
      dataSet={Object {}}
      isIndeterminate={false}
      style={
        Object {
          "backgroundColor": "#a5f3fc",
          "borderRadius": 9999,
          "height": 24,
          "overflow": "hidden",
        }
      }
    >
      <View
        dataSet={Object {}}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#0891b2",
            "display": "flex",
            "elevation": 1,
            "height": "100%",
            "justifyContent": "center",
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.18,
            "shadowRadius": 1,
            "width": "1%",
          }
        }
      />
    </View>
    <View
      className="sc-dkPtRN ioVHrU"
      dataSet={Object {}}
      mTop={50}
      style={Object {}}
    >
      <View
        _text={
          Object {
            "color": "darkText",
          }
        }
        accessibilityRole="button"
        accessible={true}
        colorScheme="red"
        dataSet={Object {}}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onHoverIn={[Function]}
        onHoverOut={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fee2e2",
            "borderRadius": 4,
            "color": "#fca5a5",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingBottom": 8,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 8,
          }
        }
        testID="reset.score.botton"
      >
        <View
          dataSet={Object {}}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            dataSet={Object {}}
            style={Object {}}
          >
            <Text
              dataSet={Object {}}
              style={
                Object {
                  "backgroundColor": undefined,
                  "color": "#ef4444",
                  "fontFamily": undefined,
                  "fontSize": 14,
                  "fontStyle": "normal",
                  "fontWeight": "500",
                  "letterSpacing": 0,
                  "lineHeight": 21,
                  "textDecorationLine": undefined,
                }
              }
            >
              Reset
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
